### PR TITLE
Removed a redundand 'resolve' in SimpleFileDialog by reusing the alre…

### LINKED ISSUE #99384: simpleFileDialog unnecessarily resolves files multiple times
- VSCode Version: Code 1.44.0 (2aae1f26c72891c399f860409176fe435a154b13)
- OS Version: Linux x64 5.2.17-1rodete3-amd64

SimpleFileDialog lists all files in a directory and creates items (createItems) by calling on each file the createItem method, which in turn calls 'resolve' a second time on the same files. Depending on the performance of 'resolve' & the file system this can significantly slow down showing files in the dialog.

### PR DESCRIPTION
**Pull request by stoyannk**
…ady queried stat data.

This PR fixes #99384

Removed the redundant call to 'resolve' when creating items in the CimpleFileDialog by reusing the stat data already available for each item.

### CODE REVIEW

File: src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts
Comment (alexr00) [inline src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts:931]: Still should do `resuorces.basename(fullPath)` before the `addTrailingPathSeparator` since you can end up with an extra path separator.

**Review by alexr00 — APPROVED** [review]
Good PR, thank you for improving the performance! We are currently getting ready for our May release, so I won't merge this PR immediately. Once we have branched for release I will merge it.

**Comment by stoyannk** [comment]
Thank you for the fast review!

**Pull request merged**
